ClientCore/ProgramConstants.cs

```diff
@@ -89,5 +103,18 @@ public static string GetAILevelName(int aiLevel)
 
         // Static fields might be initialized before the translation file is loaded. Change to readonly properties here.
         public static List<string> AI_PLAYER_NAMES => new List<string> { "Easy AI".L10N("UI:Main:EasyAIName"), "Medium AI".L10N("UI:Main:MediumAIName"), "Hard AI".L10N("UI:Main:HardAIName") };
+
+        public static string LogFileName { get; set; }
+
+        public static Action<string, string> UserErrorAction { get; set; } = (title, error) =>
```

Comment:
`ErrorHandlerAction` disagreed as it does not *handle* errors, it logs errors.
`DisplayErrorAction` disagreed as it is for displaying related errors by convention.
I prefer name it as `ErrorLogger`.

Reply:
It's not 

> `ErrorHandlerAction` disagreed as it does not _handle_ errors, it logs errors. `DisplayErrorAction` disagreed as it is for display-related errors by convention. I prefer name it as `ErrorLogger`.

Its main function is not logging but showing a message to the user. Logging is done by another function.

The default value for the Action<> is indeed writing the message to file and then opening that file but only because the UI system has not yet been loaded.
As soon as the UI system is loaded the Action<> is set to XNAMessageBox().Show() without doing any logging at all.

The invocation of this Action<> is always followed by Environment.Exit().